src/ToricVarieties/ToricDivisors/constructors.jl

```diff
@@ -67,88 +81,9 @@ export DivisorOfCharacter
 
 
 ######################
-# 5: Attributes
-######################
-
-@doc Markdown.doc"""
-    polyhedron(td::ToricDivisor)
-
-Construct the polyhedron $P_D$ of a torus invariant divisor $D:=td$ as in 4.3.2
-of [CLS11](@cite). The lattice points of this polyhedron correspond to the
-global sections of the divisor.
-
-# Examples
-The polyhedron of the divisor with all coefficients equal to zero is a point,
-if the ambient variety is complete. Changing the coefficients corresponds to
-moving hyperplanes. One direction moves the hyperplane away from the origin,
-the other moves it across. In the latter case there are no global sections
-anymore and the polyhedron becomes empty.
-```
-julia> H = hirzebruch_surface(4)
-A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
-
-julia> td0 = ToricDivisor(H, [0,0,0,0])
-A torus invariant divisor on a normal toric variety
-
-julia> isfeasible(polyhedron(td0))
-true
-
-julia> dim(polyhedron(td0))
-0
-
-julia> td1 = ToricDivisor(H, [1,0,0,0])
-A torus invariant divisor on a normal toric variety
-
-julia> isfeasible(polyhedron(td1))
-true
-
-julia> td2 = ToricDivisor(H, [-1,0,0,0])
-A torus invariant divisor on a normal toric variety
-
-julia> isfeasible(polyhedron(td2))
-false
-```
-"""
-function polyhedron(td::ToricDivisor)
-    pmtd = pm_tdivisor(td)
-    return Polyhedron(pmtd.SECTION_POLYTOPE)
-end
-export polyhedron
-
-
-@doc Markdown.doc"""
-    coefficients(td::ToricDivisor)
-
-Identify the coefficients of a toric divisor in the group of torus invariant Weil divisors.
-
-# Examples
-```
-julia> H = hirzebruch_surface(4)
-A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
-
-julia> D = ToricDivisor(H, [1,2,3,4])
-A torus invariant divisor on a normal toric variety
-
-julia> coefficients(D)
-4-element Vector{Int64}:
- 1
- 2
- 3
- 4
-```
-"""
-function coefficients(td::ToricDivisor)
-    return Vector{Int}(Polymake.common.primitive(pm_tdivisor(td).COEFFICIENTS))
-end
-export coefficients
+### 4: Display
+######################s
 
-
-###############################################################################
-###############################################################################
-### 5: Display
-###############################################################################
-###############################################################################
 function Base.show(io::IO, td::ToricDivisor)
-    print(io, "A torus invariant divisor on a normal toric variety")
+    print(io, output_string(td))
```

Comment:
why this way? Why not instead print into a string when you really need one?

Reply:
My thoughts were no deeper than extending the existing code. As always, I am happy to change to a more robust/preferred/quicker alternative. Are saying that `print(output_string(td))` is better?

Reply:
No, I am saying that instead  of shoving data into a string and then printing it, it is better to directly print it. And if you sometimes really want a string, you can simply print into a string.

Notw that even `join` takes an io argument and can thus print instead of creating a (temporary) string

Reply:
https://github.com/oscar-system/Oscar.jl/pull/839/commits/8aafd56a088b0c10e337fd2dadceb9d2490f4880 and https://github.com/oscar-system/Oscar.jl/pull/839/commits/c498e55cd7655cdb2178475ba0f2a349b316979c have been adjusted accordingly. Let me know what you think.